pkg/kinflate/app/application.go

```diff
@@ -39,8 +47,102 @@ func New(loader loader.Loader) (Application, error) {
 }
 
 // Resources computes and returns the resources from the manifest.
-func (a *applicationImpl) Resources() []resource.Resource {
+func (a *applicationImpl) Resources() (types.ResourceCollection, error) {
 	// This is where all the fun happens.
 	// From the manifest create the resources.
-	return nil
+	m, err := newManifest(a.loader)
+	if err != nil {
+		return nil, err
+	}
+	a.manifest = *m
+
+	errs := &interror.ManifestErrors{}
+	allResourcesCollection, err := rescourceCollectionFromManifest(a.loader, m)
+	if err != nil {
+		return allResourcesCollection, err
+	}
+
+	for _, pkgPath := range m.Packages {
+		subloader, err := a.loader.New(pkgPath)
+		if err != nil {
+			errs.Append(err)
+			continue
+		}
+		subapp, err := New(subloader)
+		if err != nil {
+			errs.Append(err)
+			continue
+		}
+		subResourcesCollection, err := subapp.Resources()
+		if err != nil {
+			errs.Append(err)
+			continue
+		}
+		types.Merge(allResourcesCollection, subResourcesCollection)
```

Comment:
Not a big deal but I I am thinking if ResourceCollection should expose method to add another resourceCollection ? because this merge is not resulting into a new resourceCollection instead modifying the first collection ?

Reply:
I agree. But it should not be in the scope of this PR. Will do it in a followup PR.